core/modules/replica/ServiceManagementRequest.h

```diff
@@ -0,0 +1,354 @@
+    static char const* requestTypeName() {
+        return "SERVICE_REQUESTS";
+    }
+    static proto::ReplicationServiceRequestType requestType() {
+        return proto::ReplicationServiceRequestType::SERVICE_REQUESTS;
+    }
+};
+struct ServiceDrainRequestPolicy {
+    static char const* requestTypeName() {
+        return "SERVICE_DRAIN";
+    }
+    static proto::ReplicationServiceRequestType requestType() {
+        return proto::ReplicationServiceRequestType::SERVICE_DRAIN;
+    }
+};
+
+/**
+  * Class ServiceManagementRequestBaseM is the base class for a family of requests
+  * managing worker-side replication service. The only variable parameter of this
+  * class is a specific type of the managemenyt request.
+  *
+  * Note that this class can't be instantiate directly. It serves as an implementation
+  * of the protocol. All final customizations and type-specific operations are
+  * provided via a generic subclass.
+  */
+class ServiceManagementRequestBaseM
+    :   public RequestMessenger {
+
+public:
+
+    /// The pointer type for instances of the class
+    typedef std::shared_ptr<ServiceManagementRequestBaseM> pointer;
+
+    // Default construction and copy semantics are prohibited
+
+    ServiceManagementRequestBaseM() = delete;
+    ServiceManagementRequestBaseM(ServiceManagementRequestBaseM const&) = delete;
+    ServiceManagementRequestBaseM& operator=(ServiceManagementRequestBaseM const&) = delete;
+
+    /// Destructor
+    ~ServiceManagementRequestBaseM() override = default;
+
+    /**
+     * Get the state of the worker-side service
+     *
+     * This method will throw exception std::logic_error if the request's primary state
+     * is not 'FINISHED' and its extended state is neither 'SUCCESS" or 'SERVER_ERROR'.
+     */
+    ServiceState const& getServiceState() const;
+
+protected:
+
+    /**
+     * Construct the request with the pointer to the services provider.
+     */
+    ServiceManagementRequestBaseM(ServiceProvider::pointer const& serviceProvider,
+                                  boost::asio::io_service& io_service,
+                                  char const*              requestTypeName,
+                                  std::string const&       worker,
+                                  lsst::qserv::proto::ReplicationServiceRequestType requestType,
+                                  std::shared_ptr<Messenger> const& messenger);
+private:
+
+    /**
+      * Implement the method declared in the base class
+      *
+      * @see Request::startImpl()
+      */
+    void startImpl() final;
+
+    /**
+     * Process the worker response to the requested operation.
+     *
+     * @param success - the flag indicating if the operation was successfull
+     * @param message - a response from the worker service (if success is 'true')
+     */
+    void analyze(bool success,
+                 proto::ReplicationServiceResponse const& message);
+
+private:
+
+    /// Request type
```

Comment:
I'll stop flagging mixed comment styles. I think they should all be one or the other tho.